omnibus/config/software/datadog-agent-integrations.rb

```diff
@@ -83,47 +80,73 @@
       copy windows_safe_path("#{project_dir}/.tuf-root.json"), windows_safe_path("#{tuf_repo_meta}/current/root.json")
     end
 
+    all_reqs_file = File.open("#{project_dir}/check_requirements.txt", 'w+')
+    # FIX THIS these dependencies have to be grabbed from somewhere
+    all_reqs_file.puts "pympler==0.5 --hash=sha256:7d16c4285f01dcc647f69fb6ed4635788abc7a7cb7caa0065d763f4ce3d21c0f"
+    all_reqs_file.puts "wheel==0.30.0 --hash=sha256:e721e53864f084f956f40f96124a74da0631ac13fbbd1ba99e8e2b5e9cafdf64"\
+    " --hash=sha256:9515fe0a94e823fd90b08d22de45d7bde57c90edce705b22f5e1ecf7e1b653c8"
+    
+    all_reqs_file.close
+
     # Install all the requirements
+    # Install all the build requirements
+     if windows?
+       pip_args = "install --require-hashes -r #{project_dir}/check_requirements.txt"
+       command "#{windows_safe_path(install_dir)}\\embedded\\scripts\\pip.exe #{pip_args}"
+     else
+       build_env = {
+         "LD_RUN_PATH" => "#{install_dir}/embedded/lib",
+         "PATH" => "#{install_dir}/embedded/bin:#{ENV['PATH']}",
+       }
+       pip "install --require-hashes -r #{project_dir}/check_requirements.txt", :env => build_env
+     end
+
+    # Set frozen requirements without pip-tools 
+    pip "freeze > #{install_dir}/constraint_requirements.txt"
+
+    # Install all the build requirements
     if windows?
-      pip_args = "install  -r #{project_dir}/check_requirements.txt"
+      pip_args = "install pip-tools==#{PIPTOOLS_VERSION}"
       command "#{windows_safe_path(install_dir)}\\embedded\\scripts\\pip.exe #{pip_args}"
     else
       build_env = {
         "LD_RUN_PATH" => "#{install_dir}/embedded/lib",
         "PATH" => "#{install_dir}/embedded/bin:#{ENV['PATH']}",
       }
-      pip "install -r #{project_dir}/check_requirements.txt", :env => build_env
+      pip "install pip-tools==#{PIPTOOLS_VERSION}", :env => build_env
     end
 
-    # Set frozen requirements
-    pip "freeze > #{install_dir}/agent_requirements.txt"
-
     # Windows pip workaround to support globs
     python_bin = "\"#{windows_safe_path(install_dir)}\\embedded\\python.exe\""
-    python_pip = "pip install -c #{windows_safe_path(install_dir)}\\agent_requirements.txt #{windows_safe_path(project_dir)}"
-    python_pip_req = "pip install -c #{windows_safe_path(install_dir)}\\agent_requirements.txt -r #{windows_safe_path(project_dir)}"
+    python_pip_no_deps = "pip install -c #{windows_safe_path(install_dir)}\\constraint_requirements.txt --no-deps #{windows_safe_path(project_dir)}"
+    python_pip_req = "pip install -c #{windows_safe_path(install_dir)}\\constraint_requirements.txt --require-hashes -r #{windows_safe_path(project_dir)}"
```

Comment:
As I mentioned in the other PR I think I would rather keep the trailing `#{windows_safe_path(project_dir)}` outside of the string and add it to the commands we call `python_pip_req` from, it's more explicit.